optimizely/event/user_event_factory.py

```diff
@@ -51,6 +51,8 @@ def create_impression_event(
 
         if variation_id and experiment_id:
             variation = project_config.get_variation_from_id_by_experiment_id(experiment_id, variation_id)
+        elif variation_id and flag_key:
```

Comment:
please add proper comments, why adding this logic.

Reply:
Done